tests/FilterMapTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Spatie\CollectionMacros\Test;
+
+use Illuminate\Support\Collection;
+
+class FilterMapTest extends TestCase
+{
+    /** @test */
+    public function it_returns_a_mapped_collection_without_empty_values()
+    {
+        $result = Collection::make([1, 2, 3, 4, 5, 6])->filterMap(function ($number) {
+            $half = $number / 2;
+
+            return is_float($half)
```

Comment:
Would it make sense to aim for the positive outcome:

```php
$quotient = $number / 2;

return is_integer($quotient)
  ? $quotient
  : null
```

In my example above I've changed `$half` to `$quotient`, as `$half` implies the quotient is not a whole number.

Reply:
Good call, updated!